### PR TITLE
23353 POC for new continuation in design

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#23353

**NOTE: merging this PR will BREAK the current continuation in application functionality, but forthcoming tickets will update the flows/statuses and get things working again. Olga is OK with this.**

*Description of changes:*

- app version = 5.12.0
- generalized page blurb functionality
- always show fee summary
- added special case for page header
- added special case for fee summary label
- added special route checks for continuation in authorization vs application
- added special continuation in authorization steps
- added special case for file and pay button label (submit vs resubmit)
- added special cases for validation in file and pay method
- added special case for no payment
- reverted Effective Date Time radio labels
- updated file upload error messages
- renamed ContinuationInBusinessHome -> ContinuationInAuthorizationPage
- added file name validation for ISO-8859-1 encoding
- renamed ContinuationAuthorization -> AuthorizationProof
- added filename ISO8859-1 encoding check
- updated misc existing business information layout, labels, etc
- removed confirmation checkbox
- cleaned up summary page
- fixed alignment on summary page
- renamed some getters
- added special case for dialog filing title
- renamed route + view + enums + getters + interfaces
- added special case for no fee when restoring draft
- added page blurbs to continuation in resource files
- separated auth step from the rest
- renumbered steps + hide stepper for step 0
- deleted obsolete dissolution resource objects
- added some store getters
- added Authorization Contact Information to step 0 page
- updated continuation in authorization page validation
- added special case to show pages errors and scroll properly
- removed draft-only checks on FED and staff payment components
- updated/added unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

*Sample Continuation Authorization ("step 0"):*

image

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: tests/unit/Actions.spec.ts [hunk 1]
```diff
@@ -111,14 +111,15 @@ describe('Actions component - Incorporation Application', () => {
...
-  it('Shows "Submit and Pay" button when filing is in Draft status', async () => {
+  it('Shows "Submit" button when filing is in Draft status', async () => {
     const localVue = createLocalVue()
     localVue.use(VueRouter)
     const router = mockRouter.mock()
     router.push({ name: 'continuation-in-review-confirm', query: { id: 'T1234567' } })
     const wrapper = shallowMount(Actions, {
       // declare computed properties to override local and store getters:
       computed: {
+        isContinuationInFiling: () => true,
```
Comment (severinbeauvais) [inline tests/unit/Actions.spec.ts:122]: This is an easy way to override a store getter.

File: tests/unit/ContinuationInBusinessHome.spec.ts
Comment (severinbeauvais) [inline tests/unit/ContinuationInBusinessHome.spec.ts:1]: This file was actually renamed "tests/unit/ContinuationInAuthorization.spec.ts". I don't know why GH isn't recognizing that.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: src/resources/Dissolution/GP.ts [hunk 2]
```diff
@@ -20,47 +22,5 @@ export const DissolutionResourceGp: DissolutionResourceIF = {
...
-          type: ItemTypes.PLACEHOLDER,
-          value: 'legal_name'
-        },
-        {
-          type: ItemTypes.TEXT,
-          value: 'and has been deposited in the Company\'s records book.'
-        }
-      ],
-      items: []
-    }
```
Comment (severinbeauvais) [inline src/resources/Dissolution/GP.ts:64]: ^^ Unused code. Must have come from initial copy-paste when implementing firm dissolution.

File: tests/unit/ContinuationInAuthorization.spec.ts
Comment (severinbeauvais) [inline tests/unit/ContinuationInAuthorization.spec.ts:1]: This is not a new file. It was previously named "tests/unit/ContinuationInBusinessHome.spec.ts". I don't know why GH isn't recognizing that.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: tests/unit/ContinuationInAuthorization.spec.ts
Comment (severinbeauvais) [inline tests/unit/ContinuationInAuthorization.spec.ts:1]: This is not a new file. It was previously named "tests/unit/ContinuationInBusinessHome.spec.ts". I don't know why GH isn't recognizing that.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: tests/unit/ContinuationInAuthorization.spec.ts
Comment (severinbeauvais) [inline tests/unit/ContinuationInAuthorization.spec.ts:1]: This is not a new file. It was previously named "tests/unit/ContinuationInBusinessHome.spec.ts". I don't know why GH isn't recognizing that.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: src/App.vue [hunk 3]
```diff
@@ -924,13 +946,19 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
     // NB: will throw if API error
     let draftFiling = await LegalServices.fetchFirstOrOnlyFiling(tempId)
 
+    // set filing type and status
     this.setFilingType(draftFiling.header.name)
     this.setFilingStatus(draftFiling.header.status)
+    // *** TODO: remove after testing
+    if (draftFiling.header.status === FilingStatus.CHANGE_REQUESTED) {
+      this.setFilingStatus(FilingStatus.APPROVED)
+    }
```
Comment (severinbeauvais) [inline src/App.vue:955]: This is a temporary override for testing. This PR breaks this filing in other ways so I think this is OK to merge into main branch (Dev, etc).

("Change Requested" is only used for continuation in applications, which isn't released to Prod yet, so this change should not affect anything else.)

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: src/components/ContinuationIn/AuthorizationProof.vue [hunk 4]
```diff
@@ -334,6 +334,12 @@ export default class ExtraproRegistration extends Mixins(DocumentMixin) {
    * @param file the file to add
    */
   async onFileSelected (file: File): Promise<void> {
+    /** Returns True if given string contains only Latin 1 (ie, ISO8859-1) characters. */
+    function isValidLatin1 (str: string): boolean {
+      // eslint-disable-next-line no-control-regex
+      return !/[^\u0000-\u00ff]/g.test(str)
+    }
```
Comment (severinbeauvais) [inline src/components/ContinuationIn/AuthorizationProof.vue:341]: I did a bit of research on how to identify filenames with invalid characters and this is the best / cleanest I could find. (Other solutions used libraries or packages and made other assumptions.)

This arose from a file that Janis was trying to upload and which caused an error while trying to upload the file.  Now the following error is displayed:

image

Sample file with invalid character in filename:
[Maybe it’s time (invalid filename).pdf](https://github.com/user-attachments/files/17215019/Maybe.it.s.time.invalid.filename.pdf)

File: src/components/ContinuationIn/SummaryBusinessHomeJurisdiction.vue [hunk 5]
```diff
@@ -228,14 +212,15 @@
             <v-col
               cols="12"
               sm="9"
+              class="file-buttons pt-2 pt-sm-0"
```
Comment (severinbeauvais) [inline src/components/ContinuationIn/SummaryBusinessHomeJurisdiction.vue:215]: This and below: fixed vertical alignment, ie

image

File: src/components/ContinuationIn/SummaryBusinessHomeJurisdiction.vue [hunk 6]
```diff
@@ -228,14 +212,15 @@
...
             >
               <!-- the director's affidavit file -->
               <template v-if="isContinuationInAffidavitRequired">
                 <v-btn
                   v-if="getExistingBusinessInfo.affidavitFileName"
                   text
                   color="primary"
-                  class="download-affidavit-btn mt-sm-n2 d-block"
+                  class="download-affidavit-btn mt-sm-n2 d-block px-2 ml-n2"
```
Comment (severinbeauvais) [inline src/components/ContinuationIn/SummaryBusinessHomeJurisdiction.vue:223]: This and above : fixed alignment, ie

image

File: src/views/ContinuationIn/ContinuationInReviewConfirm.vue [hunk 7]
```diff
@@ -73,7 +73,6 @@
 
     <!-- Continuation Effective Date and Time -->
     <section
-      v-if="(getFilingStatus === FilingStatus.DRAFT) || getEffectiveDateTime.isFutureEffective"
```
Comment (severinbeauvais) [inline src/views/ContinuationIn/ContinuationInReviewConfirm.vue:76]: Previously we hid some sections if this was a Draft. Now, draft applications only see "step 0". In other states, we can show these sections normally.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-734-k2qli397.web.app

File: tests/unit/ContinuationInAuthorization.spec.ts
Comment (severinbeauvais) [inline tests/unit/ContinuationInAuthorization.spec.ts:1]: This is not a new file. It was previously named "tests/unit/ContinuationInBusinessHome.spec.ts". I don't know why GH isn't recognizing that.

**Review by vysakh-menon-aot — APPROVED** [review]

**Comment by severinbeauvais** [comment]
If you have any further comments on this PR, I will fix them as part of bcgov/entity#23514.

**Pull request merged**